File: frontendRefineria/src/models/Ruta.ts
import { Surtidor } from "./Surtidor";

export interface Ruta {
    id:                 number;
    nombre:             string;
    fecha:              Date;
    camion:             Camion;
    litros_combustible: number;
    tipo_combustible:   string;
    precio_por_litro:   number;
    completado:         boolean;
    surtidores:         SurtidorSimple[] | Surtidor[];
}

export interface Camion {
    id:               number;
    nombre:           string;
    capacidad:        number;
    tipo_combustible: string;
    user_id:          number;
}

export interface SurtidorSimple {
    surtidor_id:    number;
    litros_entrega: number;
}

File: backendRefineria/refineria/models/ruta.py
from django.db import models

from refineria.models import Camion


class Ruta(models.Model):
    nombre = models.CharField(max_length=100)
    fecha = models.DateField()
    camion = models.ForeignKey(Camion, on_delete=models.CASCADE)
    litros_combustible = models.FloatField()
    tipo_combustible = models.CharField(max_length=50)
    precio_por_litro = models.FloatField()
    completado = models.BooleanField(default=False)

    def __str__(self):
        return self.nombre